### PR TITLE
Provide a baseURI polyfill

### LINKED ISSUE #964: baseURI polyfill is missing in v2
The baseURI polyfill used to be included in HTMLImports, but now that those are gone, the baseURI polyfill is too. This breaks certain elements that relied on it e.g. iron-location for IE

### PR DESCRIPTION
**Pull request by azakus**
This was lost from v1 when HTML Imports was dropped

Fixes #964

### CODE REVIEW

**Comment by azakus** [comment]
@TimvdLippe I couldn't find a reference to an existing test in webcomponentsjs. Maybe that one is in polymer?

File: src/baseuri.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+if (!native_baseURI) {
+  Object.defineProperty(Node.prototype, 'baseURI', {
+    /**
+     * @this {Node}
+     * @return {string}
+     */
+    get() {
+      // this.ownerDocument is `null` for documents
+      const doc = this.ownerDocument || this;
+      const base = /** @type {HTMLBaseElement} */ (doc.querySelector('base'));
```
Comment (bicknellr) [inline src/baseuri.js]: I think you might want `base[href]` for the selector here.

File: entrypoints/webcomponents-sd-ce-pf-index.js [hunk 2]
```diff
@@ -23,4 +23,5 @@ import '../node_modules/@webcomponents/shadydom/src/shadydom.js';
 import '../node_modules/@webcomponents/custom-elements/src/custom-elements.js';
 import '../node_modules/@webcomponents/shadycss/entrypoints/scoping-shim.js';
 import '../node_modules/@webcomponents/url/url.js';
+import '../src/baseuri.js';
```
Comment (bicknellr) [inline entrypoints/webcomponents-sd-ce-pf-index.js]: Is this not meant to be imported in other entrypoints?

**Comment by TimvdLippe** [comment]
@azakus The `document.baseURI` was previously used in `tests/url.html` and was removed in https://github.com/webcomponents/webcomponentsjs/commit/d10de4d6fe08cab97d7937d2de1a7bfede593b1c

**Comment by azakus** [comment]
Ah yeah, I'll revert that test change

File: entrypoints/webcomponents-sd-ce-pf-index.js (hunk 2 above)
Comment (azakus) [reply entrypoints/webcomponents-sd-ce-pf-index.js]: No, this is only needed for IE 11

File: src/baseuri.js (hunk 1 above)
Comment (azakus) [reply src/baseuri.js]: Good call

**Pull request merged**